### PR TITLE
functor task #1 Tetiana Dobrovolska

### PR DESCRIPTION
**Pull request by TetianaDobrovolska**

### CODE REVIEW

File: functors task (1)/student/include/impl.h [hunk 1]
```diff
@@ -8,6 +8,25 @@
  */
 struct BeerOrganizer
 {
+    BeerBrand operator()()
+    {
+        if (currentIndex == (int)BeerBrand::Max - 1)
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:13]: **Never use c-style cast in c++ code!**

File: functors task (1)/student/include/impl.h [hunk 2]
```diff
@@ -8,6 +8,25 @@
...
+        {
+            currentIndex = 0;
+        }
+        return beerStock[currentIndex++];
+    }
+private:
+    int currentIndex = {0};
+    std::array<BeerBrand, (int)BeerBrand::Max> beerStock
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:21]: here too

File: functors task (1)/student/include/impl.h [hunk 3]
```diff
@@ -40,9 +62,22 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(const AlcoholDrink adrink, const NonAlcoholDrink nadrink)
     {
+        if (AlcoholDrink::Gin == adrink && NonAlcoholDrink::LimeJuice == nadrink)
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:67]: you can simplify these checks with switch

File: functors task (1)/student/include/impl.h [hunk 4]
```diff
@@ -40,9 +62,22 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(const AlcoholDrink adrink, const NonAlcoholDrink nadrink)
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:65]: _don't use abbreviations_

File: functors task (1)/student/include/impl.h (hunk 2 above)
Comment (TetianaDobrovolska) [reply functors task (1)/student/include/impl.h]: correct

File: functors task (1)/student/include/impl.h (hunk 1 above)
Comment (TetianaDobrovolska) [reply functors task (1)/student/include/impl.h]: correct

File: functors task (1)/student/include/impl.h (hunk 3 above)
Comment (TetianaDobrovolska) [reply functors task (1)/student/include/impl.h]: done it

File: functors task (1)/student/include/impl.h (hunk 4 above)
Comment (TetianaDobrovolska) [reply functors task (1)/student/include/impl.h]: correct

File: functors task (1)/student/include/impl.h [hunk 5]
```diff
@@ -8,6 +8,25 @@
...
+    BeerBrand operator()()
+    {
+        if (currentIndex == static_cast<int>(BeerBrand::Max) - 1)
+        {
+            currentIndex = 0;
+        }
+        return beerStock[currentIndex++];
+    }
+private:
+    int currentIndex = {0};
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:20]: _prefer size_t_

**Pull request closed without merging**